src/java/fr/paris/lutece/plugins/workflow/modules/appointment/provider/AppointmentNotificationMarkers.java

```diff
@@ -0,0 +1,189 @@
+        collectionNotifyMarkers.add( createMarkerDescriptions( AppointmentWorkflowConstants.MARK_REFERENCE, AppointmentWorkflowConstants.DESCRIPTION_MARK_REFERENCE ) );
+        collectionNotifyMarkers.add( createMarkerDescriptions( AppointmentWorkflowConstants.MARK_URL_CANCEL, AppointmentWorkflowConstants.DESCRIPTION_MARK_URL_CANCEL ) );
+        collectionNotifyMarkers.add( createMarkerDescriptions( AppointmentWorkflowConstants.MARK_URL_REPORT, AppointmentWorkflowConstants.DESCRIPTION_MARK_URL_REPORT ) );
+        collectionNotifyMarkers.add( createMarkerDescriptions( AppointmentWorkflowConstants.MARK_CANCEL_MOTIF, AppointmentWorkflowConstants.DESCRIPTION_MARK_CANCEL_MOTIF ) );
+        collectionNotifyMarkers.add( createMarkerDescriptions( AppointmentWorkflowConstants.MARK_RECAP, AppointmentWorkflowConstants.DESCRIPTION_MARK_RECAP ) );
+        // Add additional markers when the notification is sent to an admin
+        if ( isAdminNotification )
+        {
+            addAdminMarkers( collectionNotifyMarkers );
+        }
+        return collectionNotifyMarkers;
+    }
+
+    /**
+     * Get a Collection of the markers and their specific values. This method is used to replace the markers by the actual values of the resource, before the
+     * notification is sent
+     * 
+     * @return a Collection of InfoMarker, made of their names and proper values
+     */
+    public Collection<InfoMarker> getMarkerValues( )
+    {
+        Collection<InfoMarker> collectionNotifyMarkers = new ArrayList<>( );
+
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_FIRSTNAME, _appointment.getFirstName( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_LASTNAME, _appointment.getLastName( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_EMAIL, _appointment.getEmail( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_DATE_APPOINTMENT, _appointment.getDateOfTheAppointment( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_TIME_APPOINTMENT, _appointment.getStartingTime( ).toString( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_END_TIME_APPOINTMENT, _appointment.getEndingTime( ).toString( ) ) );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_REFERENCE, _appointment.getReference( ) ) );
+        // Retrieve the URLs to cancel, report... an appointment
+        String strUrlReport = AppointmentApp.getReportAppointmentUrl( _appointment );
+        collectionNotifyMarkers.add( createMarkerValues( AppointmentWorkflowConstants.MARK_URL_REPORT, strUrlReport.replaceAll( "&", "&amp;" ) ) );
```

Comment:
this should be avoid, it's better to format the url in the template (this should not be formated in a href element for example)